src/main/java/org/zalando/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -8,22 +15,73 @@
 import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 @Component
 @Profile("!test")
 public class KafkaFactory {
+
     private final KafkaLocationManager kafkaLocationManager;
-    private final KafkaProducer<String, String> kafkaProducer;
+    @Nullable
+    private Producer<String, String> activeProducer;
+    private final Map<Producer<String, String>, AtomicInteger> useCount = new ConcurrentHashMap<>();
```

Comment:
AFAIU, it is possible to replace Map with AtomicInteger because you use only one producer all the time. What do you think ?